functests/1_performance/hugepages.go

```diff
@@ -57,14 +59,39 @@ var _ = Describe("[performance]Hugepages", func() {
 				hugepagesSize, err := machineconfig.GetHugepagesSizeKilobytes(page.Size)
 				Expect(err).ToNot(HaveOccurred())
 
-				availableHugepagesFile := fmt.Sprintf("/sys/devices/system/node/node%d/hugepages/hugepages-%skB/nr_hugepages", *page.Node, hugepagesSize)
+				availableHugepagesFile := fmt.Sprintf("/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages", hugepagesSize)
+				nrHugepages := checkHugepagesStatus(availableHugepagesFile, workerRTNode)
+
+				freeHugepagesFile := fmt.Sprintf("/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages", hugepagesSize)
+				freeHugepages := checkHugepagesStatus(freeHugepagesFile, workerRTNode)
+
+				Expect(int32(nrHugepages)).To(Equal(page.Count), "The number of available hugepages should be equal to the number in performance profile")
+				Expect(nrHugepages).To(Equal(freeHugepages), "On idle system the number of available hugepages should be equal to free hugepages")
+			}
+		})
+	})
+
+	Context("with multiple sizes", func() {
+		It("should be supported and available for the container usage", func() {
+			for _, page := range profile.Spec.HugePages.Pages {
+				hugepagesSize, err := machineconfig.GetHugepagesSizeKilobytes(page.Size)
+				Expect(err).ToNot(HaveOccurred())
+
+				availableHugepagesFile := fmt.Sprintf("/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages", hugepagesSize)
+				if page.Node != nil {
+					availableHugepagesFile = fmt.Sprintf("/sys/devices/system/node/node%d/hugepages/hugepages-%skB/nr_hugepages", *page.Node, hugepagesSize)
+				}
 				nrHugepages := checkHugepagesStatus(availableHugepagesFile, workerRTNode)
 
 				if discovery.Enabled() && nrHugepages != 0 {
 					Skip("Skipping test since other guests might reside in the cluster affecting results")
 				}
 
-				freeHugepagesFile := fmt.Sprintf("/sys/devices/system/node/node%d/hugepages/hugepages-%skB/free_hugepages", *page.Node, hugepagesSize)
+				freeHugepagesFile := fmt.Sprintf("/sys/kernel/mm/hugepages/hugepages-%skB/free_hugepages", hugepagesSize)
```

Comment:
Not for now, but it seems MCO is in charge of setting the huge pages, maybe we should only check the MCO profile.

Reply:
I think we can leave it for the discussions that you sent via email ;)

Reply:
definitely